config_examples/redaction.yaml

```diff
@@ -0,0 +1,42 @@
+receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: 0.0.0.0:4317
+      http:
+        endpoint: 0.0.0.0:4318
+
+processors:
+  redaction:
+    allow_all_keys: false
+    allowed_keys:
+      - description
+      - group
+      - name
+    ignored_keys:
+      - safe_attribute
+    blocked_values:
+      - "4[0-9]{12}(?:[0-9]{3})?" ## Visa credit card number
+      - "(5[1-5][0-9]{14})"       ## MasterCard number
+      - "^((25[0-5]|(2[0-4]|1\\d|[1-9]|)\\d)\\.?\b){4}$" ## IPv4 addresses
+      - "^[\\w\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$" ## email addresses
```

Comment:
`.co.uk`, `com.au` etc. are covered by this regex. `.engineering` sound like a very specific mail server, which of course can be covered by regex, but I think this is already out of the scope of standard email address format

Reply:
this is just sample config anyways, i wouldn't worry too much about it as long as the most commonly used ones are covered

Reply:
if we want to support `.engineering` or longer TLDs, we can just adapt `{2,4}` to `{2,12}`, but since it's an example user config, I don't think it's completely needed, since most of the standard TLDs are max 4 chars

Reply:
Yeah it's probably pretty tricky to get a regex together that's specific enough to only cover email addresses but permissive enough to cover all possible ones.

Then let's maybe keep this one for now but clearly mark it as simplified for demonstration purpose only:

```suggestion
      - "^[\\w\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$" ## email addresses (simplified example)
```